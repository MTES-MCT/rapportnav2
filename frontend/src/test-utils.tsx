import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
