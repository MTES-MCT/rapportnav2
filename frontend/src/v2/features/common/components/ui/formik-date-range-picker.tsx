import Text from '@common/components/ui/text.tsx'
import { FormikDatePicker, FormikDatePickerWithDateDateProps, FormikEffect, Label, THEME } from '@mtes-mct/monitor-ui'
import { FieldProps, Formik, FormikErrors } from 'formik'
import { isEmpty, isEqual } from 'lodash'
import { useEffect, useState } from 'react'
import { Stack } from 'rsuite'
import styled from 'styled-components'
import { simpleDateRangeValidationSchema } from '../../../mission-action/validation-schema/date-validation'

type DateInput = {
  startDateTimeUtc: Date
  endDateTimeUtc: Date
}

type FormikDateRangePickerProps = FormikDatePickerWithDateDateProps & {
  name: string
  validateOnSubmit?: boolean
  fieldFormik: FieldProps<Date[]>
}

export const FormikDateRangePicker = styled(
  ({ name, fieldFormik, validateOnSubmit, ...props }: FormikDateRangePickerProps) => {
    const [errors, setErrors] = useState<FormikErrors<DateInput>>()
    const [initValue, setInitValue] = useState<DateInput>()

    useEffect(() => {
      if (!fieldFormik?.field?.value) return
      const startDateTimeUtc = fieldFormik.field.value[0]
      const endDateTimeUtc = fieldFormik.field.value[1]
      setInitValue({ startDateTimeUtc, endDateTimeUtc })
    }, [fieldFormik])

    const handleSubmit = async (value: DateInput, errors: FormikErrors<DateInput>) => {
      setErrors(errors)
      if (isEqual(value, initValue)) return
      if (!isEmpty(errors) && validateOnSubmit) fieldFormik.form.setErrors({ ...fieldFormik.form.errors, ...errors })
      await fieldFormik.form.setFieldValue(name, [value.startDateTimeUtc, value.endDateTimeUtc])
    }

    return (
      <>
        {initValue && (
          <Formik
            initialValues={initValue}
            onSubmit={handleSubmit}
            enableReinitialize
            validateOnChange={true}
            validationSchema={simpleDateRangeValidationSchema}
          >
            {({ validateForm }) => (
              <>
                <FormikEffect
                  onChange={async nextValue =>
                    validateForm(nextValue).then(errors => handleSubmit(nextValue as DateInput, errors))
                  }
                />
                <Stack direction={'column'} alignItems={'flex-start'}>
                  <Stack.Item>
                    <Stack direction="row">
                      <Stack.Item>
                        <Label>{props.label || 'Date et heure de d√©but et de fin (utc)'}</Label>
                      </Stack.Item>
                      <Stack.Item>
                        <Label
                          as={'h3'}
                          style={{ marginLeft: 2, color: THEME.color.maximumRed, fontWeight: 'initial' }}
                        >
                          *
                        </Label>
                      </Stack.Item>
                    </Stack>
                  </Stack.Item>
                  <Stack.Item>
                    <Stack direction={'row'} spacing={'0.5rem'}>
                      <Stack.Item>
                        <FormikDatePicker
                          {...props}
                          label={''}
                          isRequired={true}
                          withTime={true}
                          isCompact={true}
                          name={'startDateTimeUtc'}
                          isErrorMessageHidden={true}
                        />
                      </Stack.Item>
                      <Stack.Item alignSelf={'center'}>
                        <Text as={'h4'}>au</Text>
                      </Stack.Item>
                      <Stack.Item>
                        <FormikDatePicker
                          {...props}
                          label={''}
                          isRequired={true}
                          withTime={true}
                          isCompact={true}
                          name={'endDateTimeUtc'}
                          isErrorMessageHidden={true}
                        />
                      </Stack.Item>
                    </Stack>
                  </Stack.Item>
                  <Stack.Item style={{ marginTop: '0.2rem' }}>
                    <Text as={'h3'} color={THEME.color.maximumRed}>
                      {<>{errors?.endDateTimeUtc ?? errors?.startDateTimeUtc ?? ''}</>}
                    </Text>
                  </Stack.Item>
                </Stack>
              </>
            )}
          </Formik>
        )}
      </>
    )
  }
)({})
