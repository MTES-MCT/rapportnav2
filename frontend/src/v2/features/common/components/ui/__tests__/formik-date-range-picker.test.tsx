import { Field, FieldProps, Formik } from 'formik'
import { vi } from 'vitest'
import { render } from '../../../../../../test-utils'
import { FormikDateRangePicker } from '../formik-date-range-picker'

const handleSubmit = vi.fn()
describe('FormikDateRangePicker', () => {
  it('should match the snapshot', () => {
    const wrapper = render(
      <Formik initialValues={{ dates: [] }} onSubmit={handleSubmit}>
        <Field name="dates">
          {(field: FieldProps<Date[]>) => (
            <FormikDateRangePicker label="" name="dates" isLight={true} fieldFormik={field} />
          )}
        </Field>
      </Formik>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
