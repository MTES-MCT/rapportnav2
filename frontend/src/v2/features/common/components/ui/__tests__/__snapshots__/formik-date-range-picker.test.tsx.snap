// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FormikDateRangePicker > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="rs-stack"
        style="align-items: flex-start; flex-direction: column;"
      >
        <div
          class="rs-stack-item"
        >
          <div
            class="rs-stack"
            style="align-items: center; flex-direction: row;"
          >
            <div
              class="rs-stack-item"
            >
              <label
                class="sc-gahYZc dVDTiJ Element-Label"
              >
                Date et heure de début et de fin (utc)
              </label>
            </div>
            <div
              class="rs-stack-item"
            >
              <h3
                class="sc-gahYZc dVDTiJ Element-Label"
                style="margin-left: 2px; color: rgb(225, 0, 15); font-weight: initial;"
              >
                *
              </h3>
            </div>
          </div>
        </div>
        <div
          class="rs-stack-item"
        >
          <div
            class="rs-stack"
            style="align-items: center; flex-direction: row; gap: 0.5rem;"
          >
            <div
              class="rs-stack-item"
            >
              <fieldset
                class="sc-DZJJV RRabk Element-Fieldset sc-jCWzJg iDDTiI Field-DatePicker sc-lehtBJ"
              >
                <div
                  class="sc-kUouGy hIVQIi Element-Fieldset__InnerBox"
                >
                  <div
                    class="sc-fvubMj gSwhwI"
                  >
                    <span
                      class="sc-jVxTAy dWfHTX"
                    >
                      <div
                        class="sc-hldWLq jZIKlG"
                      >
                        <div>
                          <input
                            aria-label="Jour"
                            class="sc-ikkVnJ duaSOH"
                            id="startDateTimeUtcDay"
                            maxlength="2"
                            pattern="\\d*"
                            placeholder="--"
                            type="text"
                          />
                          /
                          <input
                            aria-label="Mois"
                            class="sc-ikkVnJ duaSOH"
                            id="startDateTimeUtcMonth"
                            maxlength="2"
                            pattern="\\d*"
                            placeholder="--"
                            type="text"
                          />
                          /
                          <input
                            aria-label="Année"
                            class="sc-ikkVnJ cHlRNR"
                            id="startDateTimeUtcYear"
                            maxlength="4"
                            pattern="\\d*"
                            placeholder="----"
                            type="text"
                          />
                        </div>
                      </div>
                    </span>
                    <span
                      class="sc-jVxTAy cdDlYr"
                    >
                      <div
                        class="sc-iQqcaB inWahK"
                      >
                        <div
                          class="sc-fcSHUR bqgKYR"
                        >
                          <div>
                            <input
                              aria-label="Heure"
                              class="sc-ikkVnJ duaSOH"
                              id="startDateTimeUtcHour"
                              maxlength="2"
                              pattern="\\d*"
                              placeholder="--"
                              type="text"
                            />
                            :
                            <input
                              aria-label="Minute"
                              class="sc-ikkVnJ duaSOH"
                              id="startDateTimeUtcMinute"
                              maxlength="2"
                              pattern="\\d*"
                              placeholder="--"
                              type="text"
                            />
                          </div>
                        </div>
                      </div>
                    </span>
                  </div>
                  <div
                    class="sc-gFqXPY dMmOYJ Field-DatePicker__CalendarPicker"
                  >
                    <div
                      class="rs-picker rs-picker-date rs-picker-default rs-picker-toggle-wrapper rs-picker-cleanable"
                    >
                      <div
                        class="rs-picker-input-group rs-input-group rs-input-group-inside"
                      >
                        <input
                          aria-haspopup="dialog"
                          autocomplete="off"
                          autocorrect="off"
                          class="rs-input"
                          id="rs-«r0»"
                          inputmode="text"
                          placeholder="yyyy-MM-dd"
                          spellcheck="false"
                          type="text"
                          value=""
                        />
                        <span
                          class="rs-input-group-addon"
                        >
                          <svg
                            aria-hidden="true"
                            aria-label="calender simple"
                            class="rs-picker-caret-icon rs-icon rs-icon"
                            data-category="time"
                            data-testid="caret"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="0 0 16 16"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M4.5 0a.5.5 0 0 1 .5.5V2h6V.5a.5.5 0 0 1 1 0V2h1.5A1.5 1.5 0 0 1 15 3.5V6H2v7.5a.5.5 0 0 0 .5.5h11a.5.5 0 0 0 .5-.5v-3a.5.5 0 0 1 1 0v3a1.5 1.5 0 0 1-1.5 1.5h-11A1.5 1.5 0 0 1 1 13.5v-10A1.5 1.5 0 0 1 2.5 2H4V.5a.5.5 0 0 1 .5-.5M14 5V3.5a.5.5 0 0 0-.5-.5H12v2zm-3 0V3H5v2zM4 5V3H2.5a.5.5 0 0 0-.5.5V5z"
                            />
                            <path
                              d="M5 8v1H4V8zM8 8H7v1h1zM10 8h1v1h-1zM5 12v-1H4v1zM7 11h1v1H7zM11 11h-1v1h1z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </fieldset>
            </div>
            <div
              class="rs-stack-item"
              style="align-self: center;"
            >
              <p
                class="sc-iRcyzz sc-kbeVdv jUHdyB boIGdG"
                font-size="11"
              >
                au
              </p>
            </div>
            <div
              class="rs-stack-item"
            >
              <fieldset
                class="sc-DZJJV RRabk Element-Fieldset sc-jCWzJg iDDTiI Field-DatePicker sc-lehtBJ"
              >
                <div
                  class="sc-kUouGy hIVQIi Element-Fieldset__InnerBox"
                >
                  <div
                    class="sc-fvubMj gSwhwI"
                  >
                    <span
                      class="sc-jVxTAy dWfHTX"
                    >
                      <div
                        class="sc-hldWLq jZIKlG"
                      >
                        <div>
                          <input
                            aria-label="Jour"
                            class="sc-ikkVnJ duaSOH"
                            id="endDateTimeUtcDay"
                            maxlength="2"
                            pattern="\\d*"
                            placeholder="--"
                            type="text"
                          />
                          /
                          <input
                            aria-label="Mois"
                            class="sc-ikkVnJ duaSOH"
                            id="endDateTimeUtcMonth"
                            maxlength="2"
                            pattern="\\d*"
                            placeholder="--"
                            type="text"
                          />
                          /
                          <input
                            aria-label="Année"
                            class="sc-ikkVnJ cHlRNR"
                            id="endDateTimeUtcYear"
                            maxlength="4"
                            pattern="\\d*"
                            placeholder="----"
                            type="text"
                          />
                        </div>
                      </div>
                    </span>
                    <span
                      class="sc-jVxTAy cdDlYr"
                    >
                      <div
                        class="sc-iQqcaB inWahK"
                      >
                        <div
                          class="sc-fcSHUR bqgKYR"
                        >
                          <div>
                            <input
                              aria-label="Heure"
                              class="sc-ikkVnJ duaSOH"
                              id="endDateTimeUtcHour"
                              maxlength="2"
                              pattern="\\d*"
                              placeholder="--"
                              type="text"
                            />
                            :
                            <input
                              aria-label="Minute"
                              class="sc-ikkVnJ duaSOH"
                              id="endDateTimeUtcMinute"
                              maxlength="2"
                              pattern="\\d*"
                              placeholder="--"
                              type="text"
                            />
                          </div>
                        </div>
                      </div>
                    </span>
                  </div>
                  <div
                    class="sc-gFqXPY dMmOYJ Field-DatePicker__CalendarPicker"
                  >
                    <div
                      class="rs-picker rs-picker-date rs-picker-default rs-picker-toggle-wrapper rs-picker-cleanable"
                    >
                      <div
                        class="rs-picker-input-group rs-input-group rs-input-group-inside"
                      >
                        <input
                          aria-haspopup="dialog"
                          autocomplete="off"
                          autocorrect="off"
                          class="rs-input"
                          id="rs-«r3»"
                          inputmode="text"
                          placeholder="yyyy-MM-dd"
                          spellcheck="false"
                          type="text"
                          value=""
                        />
                        <span
                          class="rs-input-group-addon"
                        >
                          <svg
                            aria-hidden="true"
                            aria-label="calender simple"
                            class="rs-picker-caret-icon rs-icon rs-icon"
                            data-category="time"
                            data-testid="caret"
                            fill="currentColor"
                            focusable="false"
                            height="1em"
                            viewBox="0 0 16 16"
                            width="1em"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M4.5 0a.5.5 0 0 1 .5.5V2h6V.5a.5.5 0 0 1 1 0V2h1.5A1.5 1.5 0 0 1 15 3.5V6H2v7.5a.5.5 0 0 0 .5.5h11a.5.5 0 0 0 .5-.5v-3a.5.5 0 0 1 1 0v3a1.5 1.5 0 0 1-1.5 1.5h-11A1.5 1.5 0 0 1 1 13.5v-10A1.5 1.5 0 0 1 2.5 2H4V.5a.5.5 0 0 1 .5-.5M14 5V3.5a.5.5 0 0 0-.5-.5H12v2zm-3 0V3H5v2zM4 5V3H2.5a.5.5 0 0 0-.5.5V5z"
                            />
                            <path
                              d="M5 8v1H4V8zM8 8H7v1h1zM10 8h1v1h-1zM5 12v-1H4v1zM7 11h1v1H7zM11 11h-1v1h1z"
                            />
                          </svg>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="rs-stack-item"
          style="margin-top: 0.2rem;"
        >
          <p
            class="sc-iRcyzz sc-fFmWCW ekIcIT hAFmCO"
            color="#E1000F"
            font-size="13"
          />
        </div>
      </div>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="rs-stack"
      style="align-items: flex-start; flex-direction: column;"
    >
      <div
        class="rs-stack-item"
      >
        <div
          class="rs-stack"
          style="align-items: center; flex-direction: row;"
        >
          <div
            class="rs-stack-item"
          >
            <label
              class="sc-gahYZc dVDTiJ Element-Label"
            >
              Date et heure de début et de fin (utc)
            </label>
          </div>
          <div
            class="rs-stack-item"
          >
            <h3
              class="sc-gahYZc dVDTiJ Element-Label"
              style="margin-left: 2px; color: rgb(225, 0, 15); font-weight: initial;"
            >
              *
            </h3>
          </div>
        </div>
      </div>
      <div
        class="rs-stack-item"
      >
        <div
          class="rs-stack"
          style="align-items: center; flex-direction: row; gap: 0.5rem;"
        >
          <div
            class="rs-stack-item"
          >
            <fieldset
              class="sc-DZJJV RRabk Element-Fieldset sc-jCWzJg iDDTiI Field-DatePicker sc-lehtBJ"
            >
              <div
                class="sc-kUouGy hIVQIi Element-Fieldset__InnerBox"
              >
                <div
                  class="sc-fvubMj gSwhwI"
                >
                  <span
                    class="sc-jVxTAy dWfHTX"
                  >
                    <div
                      class="sc-hldWLq jZIKlG"
                    >
                      <div>
                        <input
                          aria-label="Jour"
                          class="sc-ikkVnJ duaSOH"
                          id="startDateTimeUtcDay"
                          maxlength="2"
                          pattern="\\d*"
                          placeholder="--"
                          type="text"
                        />
                        /
                        <input
                          aria-label="Mois"
                          class="sc-ikkVnJ duaSOH"
                          id="startDateTimeUtcMonth"
                          maxlength="2"
                          pattern="\\d*"
                          placeholder="--"
                          type="text"
                        />
                        /
                        <input
                          aria-label="Année"
                          class="sc-ikkVnJ cHlRNR"
                          id="startDateTimeUtcYear"
                          maxlength="4"
                          pattern="\\d*"
                          placeholder="----"
                          type="text"
                        />
                      </div>
                    </div>
                  </span>
                  <span
                    class="sc-jVxTAy cdDlYr"
                  >
                    <div
                      class="sc-iQqcaB inWahK"
                    >
                      <div
                        class="sc-fcSHUR bqgKYR"
                      >
                        <div>
                          <input
                            aria-label="Heure"
                            class="sc-ikkVnJ duaSOH"
                            id="startDateTimeUtcHour"
                            maxlength="2"
                            pattern="\\d*"
                            placeholder="--"
                            type="text"
                          />
                          :
                          <input
                            aria-label="Minute"
                            class="sc-ikkVnJ duaSOH"
                            id="startDateTimeUtcMinute"
                            maxlength="2"
                            pattern="\\d*"
                            placeholder="--"
                            type="text"
                          />
                        </div>
                      </div>
                    </div>
                  </span>
                </div>
                <div
                  class="sc-gFqXPY dMmOYJ Field-DatePicker__CalendarPicker"
                >
                  <div
                    class="rs-picker rs-picker-date rs-picker-default rs-picker-toggle-wrapper rs-picker-cleanable"
                  >
                    <div
                      class="rs-picker-input-group rs-input-group rs-input-group-inside"
                    >
                      <input
                        aria-haspopup="dialog"
                        autocomplete="off"
                        autocorrect="off"
                        class="rs-input"
                        id="rs-«r0»"
                        inputmode="text"
                        placeholder="yyyy-MM-dd"
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <span
                        class="rs-input-group-addon"
                      >
                        <svg
                          aria-hidden="true"
                          aria-label="calender simple"
                          class="rs-picker-caret-icon rs-icon rs-icon"
                          data-category="time"
                          data-testid="caret"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M4.5 0a.5.5 0 0 1 .5.5V2h6V.5a.5.5 0 0 1 1 0V2h1.5A1.5 1.5 0 0 1 15 3.5V6H2v7.5a.5.5 0 0 0 .5.5h11a.5.5 0 0 0 .5-.5v-3a.5.5 0 0 1 1 0v3a1.5 1.5 0 0 1-1.5 1.5h-11A1.5 1.5 0 0 1 1 13.5v-10A1.5 1.5 0 0 1 2.5 2H4V.5a.5.5 0 0 1 .5-.5M14 5V3.5a.5.5 0 0 0-.5-.5H12v2zm-3 0V3H5v2zM4 5V3H2.5a.5.5 0 0 0-.5.5V5z"
                          />
                          <path
                            d="M5 8v1H4V8zM8 8H7v1h1zM10 8h1v1h-1zM5 12v-1H4v1zM7 11h1v1H7zM11 11h-1v1h1z"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
          <div
            class="rs-stack-item"
            style="align-self: center;"
          >
            <p
              class="sc-iRcyzz sc-kbeVdv jUHdyB boIGdG"
              font-size="11"
            >
              au
            </p>
          </div>
          <div
            class="rs-stack-item"
          >
            <fieldset
              class="sc-DZJJV RRabk Element-Fieldset sc-jCWzJg iDDTiI Field-DatePicker sc-lehtBJ"
            >
              <div
                class="sc-kUouGy hIVQIi Element-Fieldset__InnerBox"
              >
                <div
                  class="sc-fvubMj gSwhwI"
                >
                  <span
                    class="sc-jVxTAy dWfHTX"
                  >
                    <div
                      class="sc-hldWLq jZIKlG"
                    >
                      <div>
                        <input
                          aria-label="Jour"
                          class="sc-ikkVnJ duaSOH"
                          id="endDateTimeUtcDay"
                          maxlength="2"
                          pattern="\\d*"
                          placeholder="--"
                          type="text"
                        />
                        /
                        <input
                          aria-label="Mois"
                          class="sc-ikkVnJ duaSOH"
                          id="endDateTimeUtcMonth"
                          maxlength="2"
                          pattern="\\d*"
                          placeholder="--"
                          type="text"
                        />
                        /
                        <input
                          aria-label="Année"
                          class="sc-ikkVnJ cHlRNR"
                          id="endDateTimeUtcYear"
                          maxlength="4"
                          pattern="\\d*"
                          placeholder="----"
                          type="text"
                        />
                      </div>
                    </div>
                  </span>
                  <span
                    class="sc-jVxTAy cdDlYr"
                  >
                    <div
                      class="sc-iQqcaB inWahK"
                    >
                      <div
                        class="sc-fcSHUR bqgKYR"
                      >
                        <div>
                          <input
                            aria-label="Heure"
                            class="sc-ikkVnJ duaSOH"
                            id="endDateTimeUtcHour"
                            maxlength="2"
                            pattern="\\d*"
                            placeholder="--"
                            type="text"
                          />
                          :
                          <input
                            aria-label="Minute"
                            class="sc-ikkVnJ duaSOH"
                            id="endDateTimeUtcMinute"
                            maxlength="2"
                            pattern="\\d*"
                            placeholder="--"
                            type="text"
                          />
                        </div>
                      </div>
                    </div>
                  </span>
                </div>
                <div
                  class="sc-gFqXPY dMmOYJ Field-DatePicker__CalendarPicker"
                >
                  <div
                    class="rs-picker rs-picker-date rs-picker-default rs-picker-toggle-wrapper rs-picker-cleanable"
                  >
                    <div
                      class="rs-picker-input-group rs-input-group rs-input-group-inside"
                    >
                      <input
                        aria-haspopup="dialog"
                        autocomplete="off"
                        autocorrect="off"
                        class="rs-input"
                        id="rs-«r3»"
                        inputmode="text"
                        placeholder="yyyy-MM-dd"
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <span
                        class="rs-input-group-addon"
                      >
                        <svg
                          aria-hidden="true"
                          aria-label="calender simple"
                          class="rs-picker-caret-icon rs-icon rs-icon"
                          data-category="time"
                          data-testid="caret"
                          fill="currentColor"
                          focusable="false"
                          height="1em"
                          viewBox="0 0 16 16"
                          width="1em"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M4.5 0a.5.5 0 0 1 .5.5V2h6V.5a.5.5 0 0 1 1 0V2h1.5A1.5 1.5 0 0 1 15 3.5V6H2v7.5a.5.5 0 0 0 .5.5h11a.5.5 0 0 0 .5-.5v-3a.5.5 0 0 1 1 0v3a1.5 1.5 0 0 1-1.5 1.5h-11A1.5 1.5 0 0 1 1 13.5v-10A1.5 1.5 0 0 1 2.5 2H4V.5a.5.5 0 0 1 .5-.5M14 5V3.5a.5.5 0 0 0-.5-.5H12v2zm-3 0V3H5v2zM4 5V3H2.5a.5.5 0 0 0-.5.5V5z"
                          />
                          <path
                            d="M5 8v1H4V8zM8 8H7v1h1zM10 8h1v1h-1zM5 12v-1H4v1zM7 11h1v1H7zM11 11h-1v1h1z"
                          />
                        </svg>
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="rs-stack-item"
        style="margin-top: 0.2rem;"
      >
        <p
          class="sc-iRcyzz sc-fFmWCW ekIcIT hAFmCO"
          color="#E1000F"
          font-size="13"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
